mitodl/mit-open
# Load learning_resource objects into open search

# What are the relevant tickets?
closes https://github.com/mitodl/mit-open/issues/38

# Description (What does it do?)
This PR loads data from learning_resources programs and courses into the open search index. 

There will be a follow up pr adding a search ui https://github.com/mitodl/mit-open/issues/75 that uses the search api


# How can this be tested?
On the main branch run `docker-compose run web ./manage.py backpopulate_xpro_data`, if you don't already have xpro courses and programs in your db

On this branch

run `docker-compose run web ./manage.py recreate_index --all`
Run 
`curl --location 'http://localhost:8063/api/v0/search/' \
--header 'Content-Type: application/json' \
--data '{  "post_filter": { "bool": { "must": [{"bool": { "should": [  { "term": {"resource_type": "program"}}]} }] }}}'`

and 
`curl --location 'http://localhost:8063/api/v0/search/' \
--header 'Content-Type: application/json' \
--data '{  "post_filter": { "bool": { "must": [{"bool": { "should": [  { "term": {"resource_type": "program"}}]} }] }}}'`

Verify that you see xpro course and program results

Run
`docker-compose run web ./manage.py backpopulate_xpro_data --delete`

Run the curl commands again and verify that no hits are returned


Run
`docker-compose run web ./manage.py backpopulate_xpro_data`

Run the curl commands again and verify that there are hits again

Also try running 

```
docker-compose run web ./manage.py update_index --all
```
And verify that there are no errors
